runtime/compilers/rillv1/parse_connector.go

```diff
@@ -26,5 +28,26 @@ func (p *Parser) parseConnector(node *Node) error {
 
 	r.ConnectorSpec.Driver = tmp.Driver
 	r.ConnectorSpec.Properties = tmp.Defaults
+	r.ConnectorSpec.PropertiesFromVariables, err = propertiesFromVariables(tmp.Defaults)
+	if err != nil {
+		return err
+	}
```

Comment:
You need to do the parsing and handle the error earlier in this function call – before the part that says:
```
	// NOTE: After calling insertResource, an error must not be returned. Any validation should be done before calling it.
```